libs/designer/src/lib/ui/panel/templatePanel/createWorkflowPanel/usePanelTabs.tsx

```diff
@@ -62,30 +63,46 @@ export const useCreateWorkflowPanelTabs = ({
     }
   }, [dispatch, mapping, existingWorkflowName, existingWorkflowNames, parametersExist, selectedTabId, kind]);
 
-  const handleCreateClick = useCallback(async () => {
-    setIsLoadingCreate(true);
-    await onCreateClick(() => setIsCreated(true));
-    setIsLoadingCreate(false);
+  const createWorkflowFromTemplate = useCallback(async () => {
```

Comment:
Nit: I think you can use `useMutation` from react-query over here, so you don't need to handle and create extra states for the isLoading and isError, etc. https://tanstack.com/query/v5/docs/framework/react/reference/useMutation